frappe/core/doctype/navbar_settings/navbar_settings.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020, Frappe Technologies and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+from frappe.model.document import Document
+from frappe import _
+
+class NavbarSettings(Document):
+	def validate(self):
+		self.validate_standard_navbar_items()
+
+	def validate_standard_navbar_items(self):
+		doc_before_save = self.get_doc_before_save()
+
+		before_save_items = [item for item in \
+			doc_before_save.help_dropdown + doc_before_save.settings_dropdown if item.is_standard]
+
+		after_save_items = [item for item in \
+			self.help_dropdown + self.settings_dropdown if item.is_standard]
+
+		if not frappe.flags.in_patch and (len(before_save_items) > len(after_save_items)):
```

Comment:
this is too complex logic. If you don't want to delete standard items, then just list them out and then loop them out.
```
non_removable_items = ["My Profile"]
for item in self.help_dropdown + self.settings_dropdown:
  if item.label in non_removable_items:
    non_removable_items.remove(item.label)

if non_removable_items:
   frappe.throw
```

Reply:
> this is too complex logic. If you don't want to delete standard items, then just list them out and then loop them out.
> 
> ```
> non_removable_items = ["My Profile"]
> for item in self.help_dropdown + self.settings_dropdown:
>   if item.label in non_removable_items:
>     non_removable_items.remove(item.label)
> 
> if non_removable_items:
>    frappe.throw
> ```

Since labels are changeable I am not sure this would work in cases where user has updated the label. This was the reason for adding the `is_standard` check.

Reply:
Ok, but this is still a hack.